File: src/HeavyIndustry/world/blocks/production/LiquidFuelHeater.java
package HeavyIndustry.world.blocks.production;

import mindustry.graphics.Pal;
import mindustry.ui.Bar;
import mindustry.world.blocks.heat.HeatProducer;

public class LiquidFuelHeater extends HeatProducer {
    public LiquidFuelHeater(String name) {
        super(name);
        hasLiquids = true;
    }

    @Override
    public void setBars() {
        super.setBars();
        removeBar("heat");
        addBar("heat", (LiquidFuelHeaterBuild entity) -> new Bar("bar.heat", Pal.lightOrange, () -> entity.heat / heatOutput));
    }

    public class LiquidFuelHeaterBuild extends HeatProducerBuild{
        public float liquidHeat(){
            if(liquids.current() != null){
                return liquids.current().flammability;
            } else {
                return 0;
            }
        }

        @Override
        public float heat(){
            return heat * liquidHeat();
        }
    }
}

File: src/heavyindustry/world/blocks/heat/LiquidFuelHeater.kt
package heavyindustry.world.blocks.heat

import arc.func.Prov
import mindustry.graphics.Pal
import mindustry.ui.Bar
import mindustry.world.blocks.heat.HeatProducer

open class LiquidFuelHeater(name: String?) : HeatProducer(name) {
    init {
        hasLiquids = true
        buildType = Prov { LiquidFuelHeaterBuild() }
    }

    override fun setBars() {
        super.setBars()
        removeBar("heat")
        addBar("heat") { entity: LiquidFuelHeaterBuild ->
            Bar("bar.heat", Pal.lightOrange) { entity.heat / heatOutput }
        }
    }

    inner class LiquidFuelHeaterBuild : HeatProducerBuild() {
        private fun liquidHeat(): Float {
            return if (liquids.current() != null) {
                liquids.current().flammability
            } else {
                0f
            }
        }

        override fun heat(): Float {
            return heat * liquidHeat()
        }
    }
}
